AdyenCard/Components/Card/CardViewControllerItemsProvider.swift

```diff
@@ -184,3 +208,20 @@ extension CardViewController {
     }
 
 }
+
+internal class ViewControllerPresenting: ViewControllerPresenter {
```

Comment:
A wrapper to have a nonoptional presenter that wraps a weak (and thus optional) presenter.
Ideally we make the presenter of the `FormAddressPickerItem` non optional in the init to force initialization with a presenter that is then held onto weakly.

Any other ideas?

Reply:
Idea is that `ViewControllerPresenting` will be destroyed upon destruction of `CardViewControllerItemsProvider` and this will break the retain cycle?

Reply:
Issue is that the presenter is a weak reference and thus it's optional when accessed later and that means I'd have to make the presenter parameter on `FormAddressPickerItem.init` optional.

By creating this wrapper I can make the presenter required on init but still have it released correctly.

Reply:
Do you think we can come up with a slightly improved naming? `ViewControllerPresenting` sounds more like a protocol and does not really explains the difference in behavior and the reasoning to create a new class.

Reply:
Yes, makes sense thanks - renamed it to `WeakReferenceViewControllerPresenter` and moved it to the `ViewControllerPresenter` file let me know what you think 👌

Reply:
Makes total sense now 👍